### DISCUSSION TITLE
Как обработать динамические роуты с помощью стандартной библиотеки http в node.js?

### DISCUSSION THREAD

**Discussion by AlexKharenko** [HowProgrammingWorks/LiveQA#64] [Q&A] [answered]
Хочу спросить как можно обработать динамические роуты с использованием дефолтной http библиотеки? Столкнулся с проблемой обработки роута: /chats/:id . Спасибо. https://github.com/AlexKharenko/ChatApp/blob/development/ChatBack/src/controllers/chat.controller.js

**Comment by tshemsedinov** [comment, marked as answer]
1. You have a collection of routes Map<string, function>
2. Parse by `/` and remove path sections while path not found. For example:
   - `/api/chats/10` -> `/api/chats` - found
   - `/api/localTime/China/Wuhan` -> `/api/localTime/China` -> `/api/localTime` - found

**Reply by Dnkfff** [reply]
UTC says Tokio🧔

**Comment by tshemsedinov** [comment]
Here you have a structure: https://github.com/AlexKharenko/ChatApp/blob/development/ChatBack/src/controllers/chat.controller.js
`{ httpVerb: { path: { ... } }` but it will be better to change it to `{ path: { httpVerb: { ... } }`
For example: `{ '/chats': { get: { ... }, post: { ... } }`

**Comment by tshemsedinov** [comment]
Here are more ideas how to do it in a rps way rather then rest way: https://github.com/HowProgrammingWorks/NodejsStarterKit
